### PR TITLE
Fix multistage turbine test

### PR DESCRIPTION
**Pull request by eslickj**
## Fixes

This fixes up the multistage steam turbine test.  It's required for new IAPWS binaries to pass, but it's also an improvement in general.  The main change is including the reheater temperature constraint before initializing the reheater.  This keeps the turbine from seeing a weird temperature.  It also cleans up a little out dated code.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py [hunk 1]
```diff
@@ -165,11 +176,11 @@ def test_initialize_calc_cf():
 
     # Set the inlet of the ip section, which is disconnected
     # here to insert reheater
-    p = 7.802e06
+    p = 1.4e06
     hin = pyo.value(iapws95.htpx(T=880 * pyo.units.K, P=p * pyo.units.Pa))
     m.fs.turb.ip_stages[1].inlet.enth_mol[0].value = hin
     # m.fs.turb.ip_stages[1].inlet.flow_mol[0].value = 25220.0
-    m.fs.turb.ip_stages[1].inlet.pressure[0].value = p
+    # m.fs.turb.ip_stages[1].inlet.pressure[0].value = p
```
Comment (andrewlee94) [inline idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py:183]: Do we need these commented lines? If so, a reason might help for future reference.

**Review by andrewlee94 — APPROVED** [review]
Looks OK to me - just one comment on some commented lines but nothing that needs to hold this up.

File: idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py (hunk 1 above)
Comment (bpaul4) [reply idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py:183]: Why did the `p` value change?

Comment (eslickj) [reply idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py]: The original was just an initial guess for initialization.  I set the inlet stage pressure ratio to 0.6, and in this case use that to calculate the inlet stage flow coefficient.  So the original pressure was really far off.  I think I did the original test in a hurry and despite the bad initialization setup, it passed and I didn't give it any more thought.

**Review by dallan-keylogic — APPROVED** [review]
LGTM

File: idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py (hunk 1 above)
Comment (bpaul4) [reply idaes/models_extra/power_generation/unit_models/helm/tests/test_turbine_multistage.py]: Okay, thanks for clarifying. I trust your judgement here.

**Review by bpaul4 — APPROVED** [review]

**Pull request merged**